cmd/internal/cli/singularity.go

```diff
@@ -319,6 +322,15 @@ var commonPlatformFlag = cmdline.Flag{
 	EnvKeys:      []string{"PLATFORM"},
 }
 
+// --authfile
+var commonAuthFileFlag = cmdline.Flag{
+	ID:           "commonAuthFileFlag",
+	Value:        &ociAuthFile,
+	DefaultValue: "",
```

Comment:
It's not clear to me why we don't default to the `syfs.DockerConf` value here?

Reply:
The reasoning is as follows: when the user did not specify an explicit value & the `~/.singularity/docker-config.json` file cannot be found, it makes sense to fall back to anonymous authentication (with an appropriate message to `sylog`). But if a user specified a path to an authfile and that authfile cannot be found, I _don't_ think the same behavior makes sense; it makes more sense to error out in this circumstance, because the user clearly wanted specifically the authentication located where (they thought) the file is.

This means we have to detect whether an `--authfile` argument was or wasn't passed. If we set the default to `syfs.DockerConf()`, we would have no way of distinguishing passing "~/.singularity/docker-config.json" as an argument (however much of an edge case that may be) from passing no argument at all.

Reply:
I still believe there is value in using the default value on the flag, as it would then allow someone to explicitly set `--authfile ''` to avoid authentication at all. That's a lot less disruptive than doing `registry logout`, and less hassle than manually creating a null authfile to use.

This is, I believe, a much more likely case that someone specifying singularity's managed `~/.singularity/docker-config.json' file themselves.